lib/utils.js

```diff
@@ -47,22 +51,34 @@ module.exports = function(
         dbProtocol = dataTrigger.protocol;
       }
 
-      // this needs to be updated since users might not have cloudant.com in there host name
-      if (dataTrigger.host.indexOf('cloudant.com') !== -1) {
-        // construct cloudant URL
-        dbURL = dbProtocol + '://' + dataTrigger.user + ':' + dataTrigger.pass + '@' + dataTrigger.host;
+      // input["accounturl"] = "https://" + host;
+      // unless specified host will default to accounturl without the https:// in front
+      var dbHost; 
+      if (dataTrigger.host) {
+    	  dbHost = dataTrigger.host;
       } else {
-        // construct couchDB URL
-        dbURL = dbProtocol + '://' + dataTrigger.host;
+    	  dbHost = dataTrigger.accounturl;
+    	  dbHost = dbHost.replace('https://','')
       }
+      
+      var connectionAgent = new RequestAgent({
+    	  maxSockets: maximumDbConnections
+      });
+
+      // both couch and cloudant should have their URLs in the username:password@host format
+      dbURL = dbProtocol + '://' + dataTrigger.user + ':' + dataTrigger.pass + '@' + dbHost;
 
       // add port if specified
       if (dataTrigger.port) {
         dbURL = dbURL + ':' + dataTrigger.port;
       }
 
-      logger.info(tid, method,'found trigger accounturl: ', dbURL);
-      nanoConnection = require('nano')(dbURL);
+      logger.info(tid, method,'found trigger url: ', dbURL);
+      //nanop = require('nano')({
+      //	'url': dbURL,
+      //	'requestDefaults': { 'agent' : connectionAgent }
+      //});
+      nanop = require('nano')(dbURL);
```

Comment:
this was never defined previously and does not seem to be used.  nanoconnection which was deleted above is still being referenced